src/vs/editor/browser/viewParts/viewCursors/viewCursor.ts

```diff
@@ -156,6 +168,15 @@ export class ViewCursor {
 			this._domNode.setLineHeight(this._lineHeight);
 			this._domNode.setHeight(this._lineHeight);
 
+			let desiredWidth = '1ch';
```

Comment:
@rebornix Computing the desiredWidth and setting it needs to happen only if the cursor style is set to block.